pkg/proxy/proxy/transparent.go

```diff
@@ -108,13 +108,18 @@ func NewTransparentProxyService(p *goproxy.ProxyHttpServer, ca *tls.Certificate,
 	return TransparentProxyService{
 		Proxy:      p,
 		Ca:         ca,
-		NetworkLog: netlog.CaptureActivityLog(p, m),
+		NetworkLog: &netlog.NetworkActivityLog{},
 		Mode:       mode,
 		Policy:     pl,
 		mx:         m,
 	}
 }
 
+// LogActivity starts recording network activity in memory.
+func (t *TransparentProxyService) LogActivity() {
+	t.NetworkLog = netlog.CaptureActivityLog(t.Proxy, t.mx)
```

Comment:
are you sure calling this again will free the underlying memory of the replaced object? won't the prior goroutine still be running, populating the object? some means of preempting that process (a context?) might also be required.

Reply:
I think this is obsolete now that there is no new function.